src/components/game.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
+import GameRow from './game-row';
+import '../styles/game.css';
+
+class Game extends Component {
+
+    generateRow(i) {
+        const row = this.props.game.board[i];
+
+        return (
+            <GameRow key={i}
+                      row={row}
+                      rowIndex={i}
+                      store={this.props.store}
```

Comment:
So passing around `store` like this is perfectly fine, but one of the nicer things about using Redux is that it provides a pattern called "connected components" that means you don't have to pass this prop around everywhere. This is actually one of the primary reasons that people like Redux - because it removes much of the boilerplate of passing a few props through a long hierarchy of components. There are docs on it [here](https://redux.js.org/basics/usage-with-react#implementing-container-components), but briefly it works something like this:

```
import { connect } from 'react-redux';
import { myAction } from './actions';

class MyComponent extends Component {
	// ...
    
    handleSomeClick() {
      // This is available on `props` not because I passed it in from the parent component,
      // but because I asked Redux to inject it in `mapDispatchToProps` below. 
	  this.props.myAction();
    }
}

// This is a function that takes in the `dispatch` method from the store and uses it in 
// several actions. Down below, using the `connect` method, I ask Redux to inject each of 
// these actions as props to my component. The nice thing about this is that I didn't have to
// pass a `store` to this component - I ask Redux to do that dirty work instead. 
const mapDispatchToProps = dispatch => {
  return { 
    myAction: () => dispatch(myAction)
  };
};

export default connect(() => {}, mapDispatchToProps)(MyComponent);
```

Reply:
Yea I had read about this pattern. Just hadnt gotten around to implementing it yet.